lib/hyacinth/adapters/storage_adapter/abstract.rb

```diff
@@ -5,26 +5,30 @@ module Adapters
     module StorageAdapter
       class Abstract
         def initialize(adapter_config = {})
+          raise 'Missing config option: uri_protocol' if adapter_config[:uri_protocol].blank?
+          @uri_protocol = adapter_config[:uri_protocol]
         end
 
-        # @param location_uri [String]
-        # @return [Boolean] true if this adapter can handle this type of location_uri
-        def handles?(location_uri)
-          location_uri.start_with?(uri_prefix)
+        def readable?
+          false
         end
 
-        # Generates a new storage location for the given identifier, ensuring that nothing currently exists at that location.
-        # @return [String] a location uri
-        def generate_new_location_uri(identifier)
-          raise NotImplementedError
+        def writable?
+          false
         end
 
-        def exists?(location_uri)
-          raise NotImplementedError
+        # @return [string] the expected prefix for a location_uri associated with this adapter
+        def uri_prefix
+          "#{@uri_protocol}://"
         end
 
-        # @return [string] the expected prefix for a location_uri associated with this adapter ('disk://', 'memory://', etc.)
-        def uri_prefix
+        # @param location_uri [String]
+        # @return [Boolean] true if this adapter can handle this type of location_uri
+        def handles?(location_uri)
+          location_uri.start_with?(uri_prefix)
```

Comment:
This is an existing bug, but: Should `handles?(nil)` return false or throw an error?

Reply:
Good catch, thanks!  I just found the same thing in the `PreservationAdapter` class and updated it there too.